shapely/ops.py

```diff
@@ -121,37 +122,15 @@ def cascaded_union(self, geoms):
 
         This method was superseded by :meth:`unary_union`.
         """
-        try:
-            L = len(geoms)
-            if isinstance(geoms, BaseMultipartGeometry):
-                geoms = geoms.geoms
-        except TypeError:
-            geoms = [geoms]
-            L = 1
-        subs = (c_void_p * L)()
-        for i, g in enumerate(geoms):
-            subs[i] = g._geom
-        collection = lgeos.GEOSGeom_createCollection(6, subs, L)
-        return geom_factory(lgeos.methods['cascaded_union'](collection))
+        return pygeos.union_all(geoms, axis=None)
```

Comment:
Is pygeos calling (deprecated) `GEOSUnionCascaded_r`? This is the same return for `unary_union` below.

Reply:
> Is pygeos calling (deprecated) GEOSUnionCascaded_r? This is the same return for unary_union below.

No, it's actually calling GEOSUnaryUnion_r, and not GEOSUnionCascaded_r (GEOSUnionCascaded_r is simply not implemented in pygeos). Note that this is similar to what Shapely does: for GEOS >= 3.3, `lgeos.methods['cascaded_union']` is actually an alias of `unary_union`, only for older GEOS versions it is using GEOSUnionCascaded (but pygeos doesn't support such old GEOS version):

https://github.com/Toblerity/Shapely/blob/73626bce8e99b3f251ad874a41d97d2384e734ce/shapely/geos.py#L838

Two potential discussion items out of this:

- What's the minimal supported GEOS version we want for Shapely 2.0? Shapely starts testing for GEOS 3.5 and up (the same for pygeos), but actually has code in shapely/geos.py going back to GEOS 3.0

- Do we actually want to deprecate the python `cascaded_union` function, instead of keeping it around as alias of `unary_union`? (although this is quite independent from all shapely 2.0 / pygeos refactors, but the deprecations in 1.8 could be an opportunity to deprecate this as well, if we want. But I have the impression it is still used quite a lot)

(again might be worth turning into dedicated issues)

Reply:
Opened https://github.com/Toblerity/Shapely/issues/1000 for the minimum supported GEOS version and https://github.com/Toblerity/Shapely/issues/1001 for the unary_union/cascaded_union aliasing